mesonbuild/mesonlib.py

```diff
@@ -911,8 +910,8 @@ def variable_replace(match: T.Match[str]) -> str:
 
 def do_mesondefine(line: str, confdata: 'ConfigurationData') -> str:
     arr = line.split()
-    if len(arr) != 2:
-        raise MesonException('#mesondefine does not contain exactly two tokens: %s' % line.strip())
+    if len(arr) < 2 and len(arr) > 3:
```

Comment:
There is no way this expression can ever be true. No array can have less than two **and** more than three elements at the same time.

Reply:
it can if a statement is only having mesondefine. I was about to remove this exception as it prevents the use of this statement;

```
#cmakedefine _FILE_OFFSET_BITS @_FILE_OFFSET_BITS@
```
with cdata defines as

configure_file(input: 'config.h.cmake.in', output : 'config.h', configuration : cdata, format : 'cmake@')

Reply:
You are not getting my point that this current if statement is pointless since it is logically impossible to ever be true. No number can be less than 2 **and** more than 3 at the same time. Are you sure the `and` isn't supposed to be an `or`?

Also, while `#cmakedefine` can indeed have more than two parameters, `#mesondefine`  is more strict and only allows the format `#mesondefine FOO`. So fixing `#cmakedefine` should not also affect `#mesondefine`.

Reply:
> You are not getting my point that this current if statement is pointless since it is logically impossible to ever be true. No number can be less than 2 **and** more than 3 at the same time. Are you sure the `and` isn't supposed to be an `or`?
I really wanted to avoid unconsciously the `assert` :wink: Fixed !
> 
> Also, while `#cmakedefine` can indeed have more than two parameters, `#mesondefine` is more strict and only allows the format `#mesondefine FOO`. So fixing `#cmakedefine` should not also affect `#mesondefine`.
Thanks for this clarification, I did not know about `mesondefine` format.